api/src/pcapi/alembic/versions/20240506T084310_571e971b5c73_drop_boost_username_and_password.py

```diff
@@ -0,0 +1,23 @@
+"""Drop "username" & "password" columns from "boost_cinema_details" table.
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "571e971b5c73"
+down_revision = "400f736caab1"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    op.drop_column("boost_cinema_details", "password")
+    op.drop_column("boost_cinema_details", "username")
+
+
+def downgrade() -> None:
+    op.add_column("boost_cinema_details", sa.Column("username", sa.TEXT(), autoincrement=False, nullable=False))
+    op.add_column("boost_cinema_details", sa.Column("password", sa.TEXT(), autoincrement=False, nullable=False))
```

Comment:
On doit d'abord supprimer les contraintes "NOT NULL" de ces colonnes, dans une migration pré-déploiement. Sinon, entre le moment où l'on déploie le nouveau code et le moment où la migration post-déploiement (de suppression des colonnes) est jouée, le code (l'ORM de SQLAlchemy) est susceptible de tenter de créer des lignes dans cette table sans fournir de valeur pour ces colonnes, ce qui sera rejeté par PostgreSQL car les colonnes existeront encore et seront NOT NULLable.

Reply:
Bien vu ! c'est corrigé